staging/src/k8s.io/client-go/features/envvar.go

```diff
@@ -94,6 +114,31 @@ func (f *envVarFeatureGates) Enabled(key Feature) bool {
 	panic(fmt.Errorf("feature %q is not registered in FeatureGates %q", key, f.callSiteName))
 }
 
+// Set sets the given feature to the given value.
+//
+// Features set via this method take precedence over
+// the features set via environment variables.
+func (f *envVarFeatureGates) Set(featureName Feature, featureValue bool) error {
+	feature, ok := f.known[featureName]
+	if !ok {
+		return fmt.Errorf("feature %q is not registered in FeatureGates %q", featureName, f.callSiteName)
+	}
+	if feature.LockToDefault && feature.Default != featureValue {
+		return fmt.Errorf("cannot set feature gate %q to %v, feature is locked to %v", featureName, featureValue, feature.Default)
+	}
+
+	newMapFromSetMethod := map[Feature]bool{}
+	for k, v := range f.getEnabledMapFromSetMethod() {
+		if k == featureName {
+			continue
+		}
+		newMapFromSetMethod[k] = v
+	}
+	newMapFromSetMethod[featureName] = featureValue
+	f.enabledViaSetMethod.Store(newMapFromSetMethod)
+	return nil
```

Comment:
this construct means that two parallel `.Set` calls will result in one winner and one loser.  If you lock just above this, you'll end up with two winners.

Reply:
Yes, you are right. We can do better here. Thanks, I'm glad you spotted the issue.